pal/integrity-check.ts

```diff
@@ -0,0 +1,30 @@
+declare const guard: unique symbol;
+
+type Guard = typeof guard;
+
+/**
+ * Checks if a PAL implementation module is compatible with its target interface module.
+ *
+ * @example
+ * If you write the following in somewhere:
+ *
+ * ```ts
+ * checkPalIntegrity<typeof import('pal-interface-module')>(
+ *   withImpl<typeof import('pal-implementation-module')>());
+ * ```
+ *
+ * you will receive a compilation error
+ * if your implementation module is not fulfil the interface module.
+ *
+ * @note This function should be easily tree-shaken.
```

Comment:
it seems typedoc don't support note tag

Reply:
It doesn't matter.

Reply:
I think it's better to change `@note` to `NOTE: `

because `@note` will confuse the engine developers' understanding of the typedoc tag